src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/json/ReadJSONStep.java

```diff
@@ -39,8 +39,15 @@
  * @author Nikolas Falco
  */
 public class ReadJSONStep extends AbstractFileOrTextStep {
+    /**
+     * If true POJO object will always be returned instead of the default net.sf.json.JSON objects. You can set this
+     * property with the System property
+     * <code>org.jenkinsci.plugins.pipeline.utility.steps.json.ReadJSONStep.returnPojo</code>.
+     */
+    public static final boolean RETURN_POJO_DEFAULT =
```

Comment:
this kind of initialization is not compatible with the tests.
to make that run (with this kind of static initialization) you can change to something like
```java
    public static boolean RETURN_POJO_DEFAULT = false;

    public static void reevaluateDefaults() {
        RETURN_POJO_DEFAULT = Boolean.parseBoolean(System.getProperty(ReadJSONStep.class.getName() + ".returnPojo"));
    }
    
    static {
        reevaluateDefaults();
    }
```

and call `ReadJSONStep.reevaluateDefaults()` where needed in tests:
* in tearDown() after the clear property
* in tests methods just after the `System.setProperty` in `returnPojoActivatedWithSystemProperty` & `returnPojoDeactivatedWithSystemProperty`

Another solution would be to use other Jenkins configuration capabilities.
Wouldn't it be cleaner to use the [environment variable]() for that ? they can also be centrally defined at different level (global, node, ...) if I am not mistaking.

Should the PR be splitted in 2 PRs, one with the first commit only so that it can be easily accepted.
And another one that will provide a global default?

Reply:
Indeed good remark about not mixing the 2 commits. I removed the second one and plan to open a new PR for that one. Maybe with a new issue also.